src/index.ts

```diff
@@ -7,14 +7,56 @@ import { isTestMode } from "./utils/env_helper";
 import { UIInteractionParams, UIInteraction } from "./events/ui_interaction";
 import { buildContexts } from "./utils/context_helper";
 
+const hash = require("object-hash");
+
 const snowplow = require("snowplow-tracker");
 const emitter = snowplow.emitter;
 const tracker = snowplow.tracker;
 const swdcTracker = <any>{};
+const ONE_HOUR_MILLIS = 1000 * 60 * 60;
 
 let lastProcessedTestEvent: any = {};
+let outgoingCodetimeEventMap: any = {};
+let pendingCodetimeEventTimer: any = undefined;
+
+const outgoingEventReconciler = (body: any) => {
+  if (body && Object.keys(body).length) {
+    try {
+      const payload = JSON.parse(body.request.body);
+      // find the codetime schema
+      const ctPayload = payload?.data.find((n:any) => n.ue_pr.includes("com.software/codetime"));
```

Comment:
what is `n` and `ue_pr`?

Reply:
ue_pr is from the default way that snowplow builds the payload at this raw level.   

But I think we'll want to do this for every event type.  not just codetime.

Reply:
need a better variable name than `n` though
![Screen Shot 2021-07-26 at 12 34 48 PM](https://user-images.githubusercontent.com/898851/127047969-d49878c0-d7fe-4ac5-a3e1-e7ece99209e9.png)

Reply:
ok, i'll update "n" to schema_metadata

Reply:
@mmclead so we should also keep track of other events that failed? I'm ok with that but just wanted to make sure. thanks.

Reply:
updated it to keep track and reconcile all events